packages/graphqlsp/src/persisted.ts

```diff
@@ -248,37 +252,29 @@ export const getDocumentReferenceFromTypeQuery = (
   filename: string,
   info: ts.server.PluginCreateInfo
 ): { node: ts.CallExpression | null; filename: string } => {
-  // We look for the references of the generic so that we can use the document
-  // to generate the hash.
-  const references = info.languageService.getReferencesAtPosition(
-    filename,
-    typeQuery.exprName.getStart()
-  );
-
-  if (!references) return { node: null, filename };
-
   const typeChecker = info.languageService.getProgram()?.getTypeChecker();
-  let found: ts.CallExpression | null = null;
-  let foundFilename = filename;
-  references.forEach(ref => {
-    if (found) return;
+  if (!typeChecker) return { node: null, filename };
+
+  // Handle EntityName (Identifier | QualifiedName)
+  let identifier: ts.Identifier | undefined;
+  if (ts.isIdentifier(typeQuery.exprName)) {
+    identifier = typeQuery.exprName;
+  } else if (ts.isQualifiedName(typeQuery.exprName)) {
+    // For qualified names like 'module.identifier', get the right-most identifier
+    identifier = typeQuery.exprName.right;
```

Comment:
nit: This seems a bit suspect. I don't remember fully what the intention here is, but I assume it's type casts? We may not be covering all edge cases here, if we do, this qualifies for making a helper and adding some comments